src/OpenTK/Platform/MacOS/HIDInput.cs

```diff
@@ -1032,6 +1036,12 @@ void IMouseDriver2.SetPosition(double x, double y)
             CG.WarpMouseCursorPosition(p);
         }
 
+        private void UpdateModifiers(ref KeyboardState state)
+        {
+            var v = Cocoa.SendUint(NSEvent, Selector.Get("modifierFlags"));
```

Comment:
Doesn't this require an actual NSEvent instance. I don't see how this could work. Also similar to Windows and X11 can we do this setting of state inside UpdateKeyboard.